url433bec1e56_sarobe_DifficultyInvestigation_tgz-pJ8-fr_inria_optimization_cmaes_examples_CMAExample1J8/src/main/ConvertToReadable.java

```diff
@@ -0,0 +1,75 @@
+package main;
+
+import common.Params;
+import common.utilities.JEasyFrame;
+
+import javax.swing.*;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.PrintWriter;
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.Scanner;
+
+public class ConvertToReadable {
+    public static void main(String[] args) throws Exception {
+
+        final int startingIndex = 52;
+        final int runs = 1;
+
+        for(int i=startingIndex; i<startingIndex + runs; i++) {
+            String directoryName = Params.dataDirectory + "run-" + i;
+
+            // get best-ship.txt
+            BufferedReader reader = new BufferedReader(new FileReader(directoryName + "/best-ship.txt"));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *reader*. The resource is referenced at lines 25, 31. The resource is closed at line 71. There are other execution paths that do not contain closure statements, for example, when BufferedReader.readLine throws an exception. Ensure *reader* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).